src/main/java/org/sopt/app/presentation/stamp/StampController.java

```diff
@@ -1,110 +1,93 @@
-  public Stamp findStampByMissionAndUserId(
-      @RequestHeader("userId") String userId,
-      @PathVariable Long missionId) {
-
-    return stampService.findStamp(userId, missionId);
-  }
-
-  /**
-   * 미션 달성을 위해 내용을 Stamp 내용 등록
-   */
-  @PostMapping("/{missionId}")
-  public ResponseEntity<?> uploadStamp(
-      @RequestHeader("userId") String userId,
-      @PathVariable Long missionId,
-      @RequestPart("stampContent") StampRequestDto stampRequestDto,
-      @RequestPart(name = "imgUrl", required = false) List<MultipartFile> multipartFiles
-  ) {
-
-    //스탬프 중복 검사체크
-    stampService.checkDuplicateStamp(userId, missionId);
+    private final StampService stampService;
 
-    List<String> imgPaths = s3Service.upload(multipartFiles);
-    Stamp uploadStamp = stampService.uploadStamp(stampRequestDto,
-        imgPaths, userId, missionId);
+    private final S3Service s3Service;
 
-    return new ResponseEntity<>(uploadStamp, getSuccessHeaders(), HttpStatus.OK);
-  }
+    private final StampResponseMapper stampResponseMapper;
 
-
-  @PutMapping("/{missionId}")
-  public ResponseEntity<?> editStamp(
-      @RequestHeader("userId") String userId,
-      @PathVariable Long missionId,
-      @RequestPart(value = "stampContent", required = false) StampRequestDto stampRequestDto,
-      @RequestPart(name = "imgUrl", required = false) List<MultipartFile> multipartFiles
-
-  ) {
-    //MultipartFile을 리스트에 넣어줬기 때문에 List 내부의 이미지파일에 isEmpty()를 적용해야 한다.
-    int checkNum = 1;
-    for (MultipartFile image : multipartFiles) {
-      if (image.isEmpty()) {
-        checkNum = 0;
-      }
+    @Operation(summary = "스탬프 조회하기")
+    @GetMapping("/{missionId}")
+    public ResponseEntity<StampResponse.Main> findStampByMissionAndUserId(
+            @RequestHeader("userId") String userId, @PathVariable Long missionId) {
+        val result = stampService.findStamp(userId, missionId);
+        val response = stampResponseMapper.of(result);
+        return ResponseEntity.status(HttpStatus.OK).body(response);
     }
 
-    StampResponseDto result = StampResponseDto.builder().build();
-    if (checkNum == 0) {
-
-      Stamp stamp = stampService.editStampContents(stampRequestDto, userId, missionId);
-      result.setStampId(stamp.getId());
-
-    } else {
-
-      List<String> imgPaths = s3Service.upload(multipartFiles);
-      Stamp uploadStamp = stampService.editStampWithImg(stampRequestDto,
-          imgPaths, userId, missionId);
+    @Operation(summary = "스탬프 등록하기")
+    @PostMapping("/{missionId}")
+    public ResponseEntity<StampResponse.Main> uploadStamp(
+            @RequestHeader("userId") String userId,
+            @PathVariable Long missionId,
+            @RequestPart("stampContent") StampRequest.RegisterStampRequest registerStampRequest,
+            @RequestPart(name = "imgUrl", required = false) List<MultipartFile> multipartFiles
+    ) {
+        stampService.checkDuplicateStamp(userId, missionId); // 스탬프 중복 검사체크
```

Comment:
```suggestion
        val isDuplicateStamp stampService.checkDuplicateStamp(userId, missionId); // 스탬프 중복 검사체크
        if (isDuplicateStamp) throw new ~~
```

이런 식으로 변경하면 logic에서 발생하는 에러들을 컨트롤하기 상대적으로 쉬워질 거 같습니다. controller만 보면 되니깐요!

Reply:
오오 좋습니다!